### PR TITLE
shared `stylelint-config-airbnb`

### PR DESCRIPTION
**Pull request by chlorophyllkid**
To be fair, most of this comes from [#23](https://github.com/airbnb/css/pull/23) and serves #13, but instead of having to merge with @ntwb (:+1:) each time, I wanted to create my own for faster development.

So what's new?
+ I mostly rewrote the package following the example of the current [airbnb/javascript](https://github.com/airbnb/javascript/) repository.
+ I also added some basic tests to the package.

Please feel free to comment or suggest improvements.

### CODE REVIEW

**Comment by ntwb** [comment]
Great idea @chlorophyllkid, a quick cursory glance at everything looks 👌 

I've now closed #56 🎉

**Comment by felixjung** [comment]
Neat, looking forward to this.

**Comment by chlorophyllkid** [comment]
I had to remove the rule `'scss/dollar-variable-pattern': '^foo'` because having to start every variable with `foo` seems a bit odd. :wink: 
If it should be part of the ruleset again, a proper regular expression should be created to fit the [naming conventions](https://github.com/airbnb/css#variables).

**Comment by chlorophyllkid** [comment]
Ok, I did a bunch of things today. Let me conclude:

I added the scss-lint file again, which I removed during the restructuring.
It took me a while to understand some of those, but in the end I added new rules to index.js, which now should hold every rule / recommendation of the current README and scss-lint. (please review) :wink: 

Since I added, changed or removed some of the rules, I also reordered them in index.js.
Finally, I split the tests in different files. The first is mostly for formatting rules and the other is more specific to scss syntax.

**Comment by SilentImp** [comment]
stylelint-config-airbnb is still unavailable in the npm
after reading this issue I don't quite get where is impedemence

**Comment by tony** [comment]
I'm excited about this. The tricky issue is airbnb apparently doesn't do stylelint in-house (#45), so I'm concerned that may cause friction with getting this in. They don't really have a way to maintain a shareable config if they don't use the linter.

So getting a yes/no response if airbnb is willing to officially support a stylelint config would be helpful. If the answer is no, it may be best to press forward with a separate repo. Aside: we could ask if stylelint organization would help, but the bigger picture is we need volunteers using stylelint that can help maintain it.

Doing some QA:

Was wondering the markdown lint stuff for. It's based off https://github.com/airbnb/javascript/tree/master/linters.

For reference, there is the layout of stylelint "shareable configs", which is pretty similar in spirit to what eslint offers:

- https://github.com/stylelint/stylelint-config-standard
- https://github.com/stylelint/stylelint-config-recommended

To try to get some more eyes on this, I'm posting an issue in stylelint/stylelint. https://github.com/stylelint/stylelint/issues/3091

**Comment by ntwb** [comment]
> So getting a yes/no response if airbnb is willing to officially support a stylelint config would be helpful. 

Accepting a PR for this was green-lighted in https://github.com/airbnb/css/issues/13#issuecomment-264952441

> Aside: we could ask if stylelint organization would help

I am one of the members of the stylelint org and created the original PR for this 😏 

> To try to get some more eyes on this, I'm posting an issue in stylelint/stylelint. stylelint/stylelint#3091

I've replied and closed that issue 😄

**Comment by tony** [comment]
Thanks for responding to https://github.com/stylelint/stylelint/issues/3091

I'm glad to know stylelint is *already* greenlighted in #13. That will make this process smoother.

**Comment by chlorophyllkid** [comment]
> but the bigger picture is we need volunteers using stylelint that can help maintain it.

I would be definitely up for this. :)

**Comment by ntwb** [comment]
I know Airbnb's shared ESlint configuration is popular and all but if all you're are after ia a _shared stylelint configuration_ with some _sane defaults_ to use in your projects then use:
• https://github.com/stylelint/stylelint-config-standard

The `stylelint-config-standard` configuration inherits:
• https://github.com/stylelint/stylelint-config-recommended

**Comment by chlorophyllkid** [comment]
I'm familiar with those, and I'm actually using `stylelint-config-standard` with some rule overrides. That wasn't the point of this PR and working towards this.
Since airbnb already acts for a lot of developers as a source of truth and good practise, I was wondering why this repository isn't an actual stylelint extention.

If airbnb decides to not merge those changes, I might end up copying a lot of those rules into my future project rulesets. :smile:

**Comment by stevenleija** [comment]
Any updates on this?

**Comment by SilentImp** [comment]
So, actually, why it's still not merged?
Help needed?

**Comment by lencioni** [comment]
Thanks for the contribution! At a glance this looks pretty good to me. @ljharb can you review this when you have a chance?

**Comment by alisonps** [comment]
Hi! Are there any updates on the status of this?

**Comment by chlorophyllkid** [comment]
No, still waiting for the review of @ljharb and @mikefowler :slightly_smiling_face:

File: packages/stylelint-config-airbnb/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ],
+    'comment-empty-line-before': [
+      'always',
+      { except: ['first-nested'] },
+    ],
+    'block-opening-brace-space-before': 'always',
+    'declaration-colon-space-after': 'always',
+    'declaration-colon-space-before': 'never',
+    'declaration-block-single-line-max-declarations': 1,
+    'declaration-property-value-blacklist': { '/^border/': ['none'] },
```
Comment (denisx) [inline packages/stylelint-config-airbnb/index.js]: what about it? why so?

Comment (chlorophyllkid) [reply packages/stylelint-config-airbnb/index.js]: You can find this recommendation in the README under CSS > Border: https://github.com/airbnb/css#border

And for the 'why' part:
[This codepen](https://codepen.io/denilsonsa/pen/LkdHh) shows why `border: 0;` is supperior to `border: none;`, if your intention in applying this rule is it to _remove_ the border on the given element.

**Comment by NickKaramoff** [comment]
...so? It's been 1.5 years. Is there even a point in waiting?

**Comment by tianzhich** [comment]
3 years later...😞

**Comment by chlorophyllkid** [comment]
I'm going to close this now ...this is clearly never going to happen. 😄 

Thank you everyone for your patience. Stay Safe! 👋

**Pull request closed without merging**